core/aws_ddk_core/resources/_dms.py

```diff
@@ -0,0 +1,450 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+from typing import Any, Dict, List, Optional, Union
+
+import aws_cdk.aws_dms as dms
+from aws_cdk.aws_iam import PolicyStatement, Role, ServicePrincipal
+from aws_ddk_core.config import Config
+from aws_ddk_core.resources.commons import BaseSchema
+from constructs import Construct
+from marshmallow import fields
+
+_logger: logging.Logger = logging.getLogger(__name__)
+
+
+class DMSEndpointConfiguration(BaseSchema):
+    """DDK DMS Endpoint Marshmallow schema."""
+
+
+class DMSReplicationTaskConfiguration(BaseSchema):
+    """DDK DMS ReplicationTask Marshmallow schema."""
+
+
+class DMSReplicationInstanceConfiguration(BaseSchema):
+    """DDK DMS ReplicationInstance Marshmallow schema."""
+
+    replication_instance_class = fields.Str()
+    allocated_storage = fields.Int()
+    multi_az = fields.Bool()
+    publicly_accessible = fields.Bool()
+
+
+class DMSEndpointS3SettingsConfiguration(BaseSchema):
+    """DDK DMS Endpoint S3 Settings Marshmallow schema."""
+
+    enable_statistics = fields.Bool()
+    max_file_size = fields.Int()
+
+
+class DMSFactory:
+    """
+    Class factory create and configure Kinesis DDK resources, including Delivery Streams.
+    """
+
+    @staticmethod
+    def endpoint_settings_s3(
+        scope: Construct,
+        id: str,
+        environment_id: str,
+        bucket_name: str,
+        bucket_folder: Optional[str] = None,
+        service_access_role_arn: Optional[str] = None,
+        external_table_definition: Optional[str] = None,
+        enable_statistics: Optional[bool] = None,
+        max_file_size: Optional[int] = None,
+        **endpoint_s3_props: Any,
+    ) -> dms.CfnEndpoint.S3SettingsProperty:
+        """
+        Create and configure DMS endpoint settings for s3.
+
+        This construct allows to configure parameters of the dms endpoint using ddk.json
+        configuration file depending on the `environment_id` in which the function is used.
+        Supported parameters are: ...
+
+        Parameters
+        ----------
+        scope : Construct
+            Scope within which this construct is defined
+        id: str
+            Identifier of the destination
+        environment_id: str
+            Identifier of the environment
+        bucket_name: str
+            The name of the S3 bucket.
+        bucket_folder: Optional[str]
+            An optional parameter to set a folder name in the S3 bucket.
+            If provided, tables are created in the path *bucketFolder* / *schema_name* / *table_name* / .
+            If this parameter isn’t specified, the path used is *schema_name* / *table_name* / .
+        service_access_role_arn: Optional[str]
+            A required parameter that specifies the Amazon Resource Name (ARN) used by
```

Comment:
A few things on this one:
1. Should we say it's required if the argument is set to optional?
2. We say the role must allow `iam:PassRole` but I don't see us doing that in the logic below?
3. Shall we say that if `None`, a role is created?

Reply:
1. Yep fixed, this is not required in our construct.
2. This is not needed, was carried over from my initial dev.
3. Added